football_dictionaries/assignment_2.py

```diff
@@ -1,2 +1,13 @@
+from .assignment_1 import players_as_dictionaries
+
 def players_by_position(squads_list):
-    pass
+    position_dict = players_as_dictionaries(squads_list)
+    player_position = {}
+    for player in position_dict:
+        pos = player["position"]
+        player_position.setdefault(pos, [])
+        player_position[pos].append(player)
+    return player_position    
+        
```

Comment:
I refactored and used a similar approach as well. Here's my example for your reference.
The only thing I did differently was use the built in method .get() to retrieve the position. (line 7. in your script.)

    formatted_list = players_as_dictionaries(squads_list)
    positions_list = {}
    
    # gets positions from all the players
    for player in formatted_list:
        position = player.get('position')
        # adds a field position
        positions_list.setdefault(position, [])
        # adds the player to their respective list
        # by position
        positions_list[position].append(player)

    return positions_list